### ISSUE TITLE
Removing moment.js to reduce total package size

### ISSUE THREAD

**Issue by bryceosterhaus** [liferay/clay#3222]
Stems from https://github.com/liferay/clay/issues/2918#issuecomment-627179814

We should aim to see how we can reduce the total size of what we are providing.

Most recent build shows our node_modules looking like
![Screen Shot 2020-05-12 at 11 20 19 AM](https://user-images.githubusercontent.com/6843530/81731896-7a5fb100-9444-11ea-8ebf-560c019e5415.png)

I think a major win would be in removing `moment` in favor of a smaller library or vanilla js.

**Comment by wincent** [comment]
[Bundlephobia is nice for visualizing this kind of thing too](https://bundlephobia.com/result?p=@clayui/date-picker@3.2.0):

![Screenshot 2020-05-13 -103005-VrCBG8PL@2x](https://user-images.githubusercontent.com/7074/81789756-b9edd200-9504-11ea-89c7-94d0f2afc4b9.png)

I know we talked about making an "omnibus" package (ie. that would basically contain all of Clay) in some other thread. If we pointed a tool like Bundlephobia at such a package that would give us an bird's eye view of the biggest offenders.
